driver-core/src/main/java/com/datastax/driver/core/Token.java

```diff
@@ -418,7 +418,8 @@ OPPToken hash(ByteBuffer partitionKey) {
             start = toBigInteger(oppStartToken.value, significantBytes);
             end = toBigInteger(oppEndToken.value, significantBytes);
             range = end.subtract(start);
-            if (addedBytes == 4 || range.compareTo(bigNumberOfSplits) >= 0) break;
+            if (addedBytes == 4 || start.equals(end) || range.compareTo(bigNumberOfSplits) >= 0)
```

Comment:
@amandachow shouldn't we add the same extra check `start.equals(end)` to the other branch of the `if` block below? I don't think so (I couldn't think of any case where this could happen), but just wanted to confirm with you.

Reply:
I think it's correct to _not_ add this condition to the other case. For example if your start is 0x10 and your end is 0x1, and you don't pad any bytes, your significant bytes is 2 even though your split param may require more than 2 bytes in the computation. Since in this wraparound condition the ring length is added to it, it's correct to increment the significant bytes until the corresponding ring length accommodates the number of requested splits

Reply:
We are on the same page then, thanks 👍